### PR TITLE
Meta Kdocs Playgrounds

### LINKED ISSUE #1163: Kelisli `@ extension` generation produces methods that return kinded values instead of applied types
Potentially an issue will all kinds with arity > 2

### PR DESCRIPTION
**Pull request by raulraja**
The following PR brings the ability to project KDoc executable code examples from type classes into their instances automatically generating concrete runnable snippets on each of the projected extensions:

ank_extensions

Additionaly it includes:

- [x] `:ank:playground`. Allows executable snippets that automatically are rendered as playgrounds
- [x] `:ank:playground:extension`. Used to document type classes that project code examples to their instances.
- [x] `Functor.void` -> `Functor.unit`. because files named `void` are ignored in javadocs by Dokka.
- [x] Fixes #1163  incorrect projection of kinded values with arity > 2

### CODE REVIEW

File: modules/core/arrow-data/src/test/kotlin/arrow/data/ValidatedTest.kt [hunk 1]
```diff
@@ -251,3 +251,26 @@ class ValidatedTest : UnitSpec() {
     }
   }
 }
+
+data class NetworkCorruptedField(val msg: String)
+data class Field(val msg: String)
+
+private fun map(list: List<Validated<NonEmptyList<NetworkCorruptedField>, Field>>): Validated<NonEmptyList<NetworkCorruptedField>, List<Field>> {
+  val errorList = ArrayList<NetworkCorruptedField>()
```
Comment (pakoito) [inline modules/core/arrow-data/src/test/kotlin/arrow/data/ValidatedTest.kt]: `mutableListOf()` so it's idiomatic Kotlin :D

File: modules/core/arrow-annotations-processor/src/main/java/arrow/meta/encoder/jvm/StringTypeExtensions.kt [hunk 2]
```diff
@@ -40,23 +41,55 @@ internal fun String.asClassy(): TypeName.Classy {
...
+      }
     }
+  }
+}
+
+fun String.quote(): String =
+  split(".").joinToString(".") {
+    if (KotlinReservedKeywords.contains(it)) "`$it`"
+    else it
   }
```
Comment (pakoito) [inline modules/core/arrow-annotations-processor/src/main/java/arrow/meta/encoder/jvm/StringTypeExtensions.kt]: Extra endline :p

File: build.gradle [hunk 3]
```diff
@@ -62,6 +62,11 @@ allprojects {
         delete pathApiDocs
     }
 
+    //attempt to reduce travis memory constrains
+    tasks.withType(Test) {
```
[the remark this answers is not in the document]
Comment (raulraja) [reply build.gradle]: Yes, forks tests and reduces memory consumption

File: modules/core/arrow-typeclasses/src/main/kotlin/arrow/typeclasses/Functor.kt [hunk 4]
```diff
@@ -2,31 +2,242 @@ package arrow.typeclasses
 
 import arrow.Kind
 import arrow.core.Tuple2
+import arrow.documented
 
 /**
  * ank_macro_hierarchy(arrow.typeclasses.Functor)
+ *
+ * The [Functor] type class abstracts the ability to `map` over the computational context of a type constructor.
```
[the remark this answers is not in the document]
Comment (nomisRev) [reply modules/core/arrow-typeclasses/src/main/kotlin/arrow/typeclasses/Functor.kt]: AFAIK it's not possible to do multiline GitHub suggestion, right?

**Pull request merged**